packages/fxa-settings/src/components/FormResetPasswordWithBalloon/index.tsx

```diff
@@ -111,24 +114,41 @@ export const FormResetPasswordWithBalloon = ({
             name="confirmPassword"
             label="Re-enter password"
             className="text-start"
-            onChange={() => trigger(['newPassword', 'confirmPassword'])}
+            onChange={() => {
+              if (passwordMatchErrorText) {
+                setPasswordMatchErrorText(undefined);
+              }
+              trigger(['newPassword', 'confirmPassword']);
+            }}
+            onBlurCb={() => {
+              if (errors.confirmPassword) {
+                setPasswordMatchErrorText('Passwords must match');
+              }
+            }}
             // Password confirmation field is disabled until new password is valid
             disabled={!formState.dirtyFields.newPassword || errors.newPassword}
+            hasErrors={
+              errors.confirmPassword && passwordMatchErrorText ? true : false
```

Comment:
Since `&&` is already a Boolean result, I don't think you need the inline `? true : false` wrapper?

```suggestion
              errors.confirmPassword && passwordMatchErrorText
```